app/src/main/java/org/dicekeys/app/viewmodels/DiceKeyViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package org.dicekeys.app.viewmodels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import org.dicekeys.app.encryption.EncryptedDiceKey
+import org.dicekeys.app.encryption.EncryptedStorage
+import org.dicekeys.app.repositories.DiceKeyRepository
+import org.dicekeys.dicekey.DiceKey
+
+
+class DiceKeyViewModel @AssistedInject constructor(val encryptedStorage: EncryptedStorage, val diceKeyRepository: DiceKeyRepository, @Assisted val keyId: String) : ViewModel() {
+    val isSaved = MutableLiveData<Boolean>(false)
+    val diceKey = MutableLiveData<DiceKey<*>>()
+
+    private val encryptedStorageObserver = Observer<List<EncryptedDiceKey>> {
+        diceKey.value?.let { diceKey ->
+            isSaved.postValue(it.find { it.keyId == diceKey.keyId } != null)
+        }
+    }
+
+    init {
+        diceKeyRepository.get(keyId)?.let {
+            diceKey.value = it
```

Comment:
If the only way to get a DiceKey into the view model is to load it from encrypted storage, doesn't that mean we have to store it in EncryptedStorage even if the user doesn't want it stored?

Why not have it initialized from a DiceKey?

Reply:
The only way you get the DiceKey into the view model is from the DiceKeyRepository. 
DiceKeyRepository has DiceKeys in memory. Maybe a DiceKey have an EncryptedDiceKey saved.

I can change the constructor of the ViewModel to accept the DiceKey from the fragment and not the keyId.

Reply:
Or document how dicekeys are added